pshops_lib/check_account_no.js

```diff
@@ -0,0 +1,27 @@
+/*
+    Function: check_account_no.js
+    Purpose: Checks if the input account number appears as a PShop account in Roster
+*/
+
+module.exports = function(accnum){
+    // Authentication against Roster
+    var rosterAPI = require('ext/Roster_v1_0_2/api');
+    catchAll(function() {
+        rosterAPI.verbose = true;
+        rosterAPI.dataTableAttach();
+
+        if (rosterAPI.authClient(accnum,'RW')) {
+            var client = rosterAPI.getClient(accnum,'RW');
+            if(client.DistrictName == "RRT P-Shops"){
+                state.vars.AccStatus = "Valid P-shop";
+                state.vars.Client = JSON.stringify(client);
+                state.vars.TotalCredit = client.BalanceHistory[0].TotalCredit;
+            state.vars.TotalRepay_Incl = client.BalanceHistory[0].TotalRepayment_IncludingOverpayments;
+            state.vars.Balance = client.BalanceHistory[0].Balance;
+            state.vars.farmer_name = client.ClientName;
+            }
+        else {
+            state.vars.AccStatus = "Non P-Shop"}
+        }
+    });
```

Comment:
a function like this typically should return a value. here you're setting a global variable to use later, which will work, but is much harder to follow and may have unintended consequences. in this case, all you need out of this function is a true / false response (is it or is it not a pshop??) so setting a global variable then (later) checking a global variable vs a hardcoded string is both more code to write, slower to process, and riskier